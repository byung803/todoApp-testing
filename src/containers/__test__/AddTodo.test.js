import React from 'react';
import { shallow } from 'enzyme';
import { AddTodo } from '../AddTodo';

let wrapper, value, addTodoSpy;

beforeEach(() => {
    addTodoSpy = jest.fn();
    wrapper = shallow(<AddTodo addTodo={addTodoSpy} />);
})

it('should have a form, input, button', () => {
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('input').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
});

describe('AddTodo container', () => {
    beforeEach(() => {
        value = 'abcd';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.update();
    })

    it('should change the value by handleInput', () => {
        expect(wrapper.find('input').prop('value')).toEqual(value);
    })

    it('should set value to empty after submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });
        wrapper.update();
        expect(wrapper.find('input').prop('value')).toEqual('');
    })

    it('should call AddTodoSpy when submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });
        wrapper.update();
        expect(addTodoSpy).toHaveBeenCalled();
    })

    it('should maintain snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
});