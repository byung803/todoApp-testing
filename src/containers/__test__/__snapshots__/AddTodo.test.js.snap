// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo container should maintain snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="abcd"
  />
  <button
    type="submit"
  >
    Add Todo
  </button>
</form>
`;
